README.md

```diff
@@ -56,8 +57,9 @@ $cache->set('foo', 'bar');
 ```
 
 This example eventually sets the value of the key `foo` to `bar`. If it
-already exists, it is overridden. No guarantees are made as to when the cache
-value is set. If the cache implementation has to go over the network to store
+already exists, it is overridden. To provide guarantees as to when the cache
+value is set a promise is returned. Which will fulfill with `true` on success or 
```

Comment:
This should either be one sentence ("...a promise is returned which will fulfill...") or better separated ("...a promise is returned. The promise will fulfill...").